src/plugins/intel_cpu/src/config.cpp

```diff
@@ -73,49 +68,68 @@ void Config::applyDebugCapsProperties() {
 void Config::readProperties(const ov::AnyMap& prop, const ModelType modelType) {
     const auto streamExecutorConfigKeys =
         streamExecutorConfig.get_property(ov::supported_properties.name()).as<std::vector<std::string>>();
-    const auto hintsConfigKeys = perfHintsConfig.SupportedKeys();
     for (const auto& kvp : prop) {
         const auto& key = kvp.first;
-        const auto& val = kvp.second.as<std::string>();
-        IE_SUPPRESS_DEPRECATED_START
+        const auto& val = kvp.second;
         if (streamExecutorConfigKeys.end() !=
             std::find(std::begin(streamExecutorConfigKeys), std::end(streamExecutorConfigKeys), key)) {
-            streamExecutorConfig.set_property(key, val);
+            streamExecutorConfig.set_property(key, val.as<std::string>());
             if (key == ov::affinity.name()) {
-                const auto affinity_val = ov::util::from_string(val, ov::affinity);
-                if (affinity_val == ov::Affinity::CORE || affinity_val == ov::Affinity::HYBRID_AWARE) {
-                    enableCpuPinning = true;
-                    changedCpuPinning = true;
-                } else if (affinity_val == ov::Affinity::NUMA) {
-                    enableCpuPinning = false;
-                    changedCpuPinning = true;
+                changedCpuPinning = true;
+                try {
+                    const auto affinity_val = val.as<ov::Affinity>();
```

Comment:
@wangleis @ilya-lavrenov  Don't we want to deprecate `affinity` property leaving `cpu_bind_thread`?

Reply:
@dmitry-gorokhov Since `affinity` has been used by customer already, we are mapping `affinity` to `enable_cpu_pinning` in master branch now. 

How about delivering deprecation message from 2024.0 and deprecating `affinity` in 2025.0?

Reply:
CVS-127314, CVS-129093